### PR TITLE
Improve MindsDB Knowledge Base Tool Robustness & Test Coverage

### PR DESCRIPTION
**Pull request by rawathemant246**
Summary
This PR:

- Fixes: #11156 ([Bug]: Agent is not able to query knowledge base)
- Addresses: SQL syntax issues with KB identifiers (backtick handling)

Fixes SQL query issues with MindsDB knowledge bases caused by backticks around identifiers.

Adds and corrects unit tests for all Knowledge Base tools.

Improves mock usage in tests to match actual code.

Updates test assertions to align with production logic.

Ensures all tests pass on Python 3.12.

What & Why
1. Fix: Auto-Remove Backticks from Table Names

Some MindsDB deployments fail if SQL queries use backticks around knowledge base names.

This PR ensures backticks are automatically stripped from SQL identifiers before query execution.

2. Testing Improvements

Unit tests now mock the correct database methods (get_usable_knowledge_base_names, run_no_throw) instead of run.

Test assertions are corrected for:

Output format (JSON strings for KB lists).

Error messages and headers in info/query tools.

Backtick-stripping logic and edge cases.

Adds regression coverage for all tools, including edge cases and invalid input.

3. No Production Logic Change (unless you opt for stricter input validation)

### CODE REVIEW

**Comment by rawathemant246** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by rawathemant246** [comment]
recheck

File: mindsdb/interfaces/skills/custom/text2sql/mindsdb_kb_tools.py [hunk 1]
```diff
@@ -223,6 +230,7 @@ def _run(self, tool_input: str) -> str:
         try:
             # Execute the query
             query = llm_str_strip(query)
+            query = self._remove_backticks_from_identifiers(query)
```
Comment (ea-rus) [inline mindsdb/interfaces/skills/custom/text2sql/mindsdb_kb_tools.py:233]: this should brake there queries that have table names and column names with special characters 
It was fixed in this PR https://github.com/mindsdb/mindsdb/pull/11101
@MinuraPunchihewa, is it correct?

Comment (MinuraPunchihewa) [reply mindsdb/interfaces/skills/custom/text2sql/mindsdb_kb_tools.py:233]: Hmm.. @ea-rus Yes I believe that since querying KBs also call the `run_no_throw()` function, the above PR handles this already.

Comment (rawathemant246) [reply mindsdb/interfaces/skills/custom/text2sql/mindsdb_kb_tools.py:233]: Hey thanks  for replying! :+1:

**Comment by martyna-mindsdb** [comment]
Hi @rawathemant246 
Are you still working on this?

**Comment by rawathemant246** [comment]
@martyna-mindsdb  Not currently i have read the above threads

**Comment by martyna-mindsdb** [comment]
Closing this PR for now due to inactivity.
